velox/functions/lib/tests/Re2FunctionsTest.cpp

```diff
@@ -384,6 +406,206 @@ TEST_F(Re2FunctionsTest, likePattern) {
   EXPECT_EQ(like("abc", "MEDIUM POLISHED%"), false);
 }
 
+TEST_F(Re2FunctionsTest, likeDeterminePatternKind) {
+  EXPECT_EQ(
+      determinePatternKind("_"), std::make_pair(PatternKind::kWildcardOne, 1));
+  EXPECT_EQ(
+      determinePatternKind("____"),
+      std::make_pair(PatternKind::kWildcardOne, 4));
+  EXPECT_EQ(
+      determinePatternKind("%"), std::make_pair(PatternKind::kWildcardTwo, 0));
+  EXPECT_EQ(
+      determinePatternKind("__%%__"),
+      std::make_pair(PatternKind::kWildcardTwo, 4));
+  EXPECT_EQ(
+      determinePatternKind("%_%%"),
+      std::make_pair(PatternKind::kWildcardTwo, 1));
+  EXPECT_EQ(
+      determinePatternKind("_b%%__"), std::make_pair(PatternKind::kGeneric, 0));
+  EXPECT_EQ(
+      determinePatternKind("%_%p"), std::make_pair(PatternKind::kGeneric, 0));
+
+  EXPECT_EQ(
+      determinePatternKind("presto"), std::make_pair(PatternKind::kFixed, 6));
+  EXPECT_EQ(
+      determinePatternKind("hello"), std::make_pair(PatternKind::kFixed, 5));
+  EXPECT_EQ(determinePatternKind("a"), std::make_pair(PatternKind::kFixed, 1));
+  EXPECT_EQ(
+      determinePatternKind("helloPrestoWorld"),
+      std::make_pair(PatternKind::kFixed, 16));
+  EXPECT_EQ(
+      determinePatternKind("aBcD_"), std::make_pair(PatternKind::kGeneric, 0));
+  EXPECT_EQ(
+      determinePatternKind("%aBc_D%"),
+      std::make_pair(PatternKind::kGeneric, 0));
+
+  EXPECT_EQ(
+      determinePatternKind("presto%"), std::make_pair(PatternKind::kPrefix, 6));
+  EXPECT_EQ(
+      determinePatternKind("hello%%"), std::make_pair(PatternKind::kPrefix, 5));
+  EXPECT_EQ(
+      determinePatternKind("a%"), std::make_pair(PatternKind::kPrefix, 1));
+  EXPECT_EQ(
+      determinePatternKind("helloPrestoWorld%%%"),
+      std::make_pair(PatternKind::kPrefix, 16));
+  EXPECT_EQ(
+      determinePatternKind("aBcD%%e%"),
+      std::make_pair(PatternKind::kGeneric, 0));
+  EXPECT_EQ(
+      determinePatternKind("aBc_D%%"),
+      std::make_pair(PatternKind::kGeneric, 0));
+
+  EXPECT_EQ(
+      determinePatternKind("%presto"), std::make_pair(PatternKind::kSuffix, 6));
+  EXPECT_EQ(
+      determinePatternKind("%%hello"), std::make_pair(PatternKind::kSuffix, 5));
+  EXPECT_EQ(
+      determinePatternKind("%a"), std::make_pair(PatternKind::kSuffix, 1));
+  EXPECT_EQ(
+      determinePatternKind("%%%helloPrestoWorld"),
+      std::make_pair(PatternKind::kSuffix, 16));
+  EXPECT_EQ(
+      determinePatternKind("%%_%aBcD"),
+      std::make_pair(PatternKind::kGeneric, 0));
+  EXPECT_EQ(
+      determinePatternKind("%%a%%BcD"),
+      std::make_pair(PatternKind::kGeneric, 0));
+}
+
+TEST_F(Re2FunctionsTest, likePatternWildcard) {
+  auto like = [&](std::optional<std::string> str,
```

Comment:
Looks like optional is not needed here.

Reply:
Thanks for the suggestion, removed optional for pattern but it is needed for str because of the evaluateOnce function definition.